src/isa/exec.rs

```diff
@@ -669,17 +669,15 @@ impl InstructionSet for BytesOp {
                     let (s1, s2) = regs.get_both_s(*reg1, *reg2)?;
                     let r1 = s1.as_ref();
                     let r2 = s2.as_ref();
-                    let len = r2.len();
+                    let r1_len = r1.len();
+                    let r2_len = r2.len();
                     let mut count = 0usize;
-                    for i in 0..r1.len() {
-                        if r1[i..len] == r2[..len] {
+                    for i in 0..(r1_len + 1).saturating_sub(r2_len) {
```

Comment:
`r1_len + 1` can also panic (in debug) or overflow (in release)

Reply:
r1_len is at most u16::MAX stored as usize. so it will not overflow on +1

Reply:
but I have to admit this is error prone. fixing it